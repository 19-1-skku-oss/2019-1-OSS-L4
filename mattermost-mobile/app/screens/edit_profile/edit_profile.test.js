// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import EditProfile from './edit_profile.js';

jest.mock('react-intl');
jest.mock('app/utils/theme', () => {
    const original = require.requireActual('app/utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('edit_profile', () => {
    const navigator = {
        setOnNavigatorEvent: jest.fn(),
        setButtons: jest.fn(),
        dismissModal: jest.fn(),
        push: jest.fn(),
    };

    const actions = {
        updateUser: jest.fn(),
        setProfileImageUri: jest.fn(),
        removeProfileImage: jest.fn(),
    };

    const baseProps = {
        actions,
        config: {
            ShowEmailAddress: true,
        },
        theme: Preferences.THEMES.default,
        navigator,
        currentUser: {
            first_name: 'Dwight',
            last_name: 'Schrute',
            username: 'ieatbeets',
            email: 'dwight@schrutefarms.com',
            nickname: 'Dragon',
            position: 'position',
        },
        commandType: 'ShowModal',
    };

    test('should match snapshot', async () => {
        const wrapper = shallow(
            <EditProfile {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.instance().renderProfilePicture()).toMatchSnapshot();
    });

    test('should match state on handleRemoveProfileImage', () => {
        const newNavigator = {
            dismissModal: jest.fn(),
            setOnNavigatorEvent: jest.fn(),
            setButtons: jest.fn(),
        };
        const wrapper = shallow(
            <EditProfile
                {...baseProps}
                navigator={newNavigator}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        wrapper.setProps({profileImageRemove: false});

        const instance = wrapper.instance();
        instance.emitCanUpdateAccount = jest.fn();
        instance.handleRemoveProfileImage();

        expect(wrapper.state('profileImageRemove')).toEqual(true);
        expect(instance.emitCanUpdateAccount).toHaveBeenCalledTimes(1);
        expect(instance.emitCanUpdateAccount).toBeCalledWith(true);

        expect(newNavigator.dismissModal).toHaveBeenCalledTimes(1);
        expect(newNavigator.dismissModal).toBeCalledWith({animationType: 'none'});
    });
});
