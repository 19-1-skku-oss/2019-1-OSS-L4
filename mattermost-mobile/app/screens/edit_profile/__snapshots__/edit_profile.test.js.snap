// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit_profile should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 25,
    }
  }
>
  <ProfilePictureButton
    blurTextBox={[Function]}
    browseFileTypes="public.image"
    canBrowseVideoLibrary={false}
    canTakeVideo={false}
    currentUser={
      Object {
        "email": "dwight@schrutefarms.com",
        "first_name": "Dwight",
        "last_name": "Schrute",
        "nickname": "Dragon",
        "position": "position",
        "username": "ieatbeets",
      }
    }
    maxFileSize={20971520}
    navigator={
      Object {
        "dismissModal": [MockFunction],
        "push": [MockFunction],
        "setButtons": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "leftButtons": Array [
                  Object {
                    "id": "close-settings",
                    "title": undefined,
                  },
                ],
                "rightButtons": Array [
                  Object {
                    "disabled": true,
                    "id": "update-profile",
                    "showAsAction": "always",
                    "title": undefined,
                  },
                ],
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "setOnNavigatorEvent": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      }
    }
    onShowFileSizeWarning={[Function]}
    onShowUnsupportedMimeTypeWarning={[Function]}
    removeProfileImage={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
    uploadFiles={[Function]}
    validMimeTypes={
      Array [
        "image/jpeg",
        "image/jpeg",
        "image/jpg",
        "image/jp_",
        "application/jpg",
        "application/x-jpg",
        "image/pjpeg",
        "image/pipeg",
        "image/vnd.swiftview-jpeg",
        "image/x-xbitmap",
        "image/png",
        "application/png",
        "application/x-png",
        "image/bmp",
        "image/x-bmp",
        "image/x-bitmap",
        "image/x-xbitmap",
        "image/x-win-bitmap",
        "image/x-windows-bmp",
        "image/ms-bmp",
        "image/x-ms-bmp",
        "application/bmp",
        "application/x-bmp",
        "application/x-win-bitmap",
      ]
    }
    wrapper={true}
  >
    <Connect(ProfilePicture)
      edit={true}
      imageUri={null}
      size={150}
      statusBorderWidth={6}
      statusSize={40}
    />
  </ProfilePictureButton>
</View>
`;
